bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingRegistryImpl.java

```diff
@@ -248,7 +248,13 @@ public Thing createThingOfType(ThingTypeUID thingTypeUID, ThingUID thingUID, Thi
         for (ThingHandlerFactory thingHandlerFactory : thingHandlerFactories) {
             if (thingHandlerFactory.supportsThingType(thingTypeUID)) {
                 Thing thing = thingHandlerFactory.createThing(thingTypeUID, configuration, thingUID, bridgeUID);
-                thing.setLabel(label);
+                if (thing == null) {
```

Comment:
This change is not related to the PR. Is this fixing something else that you just witnessed? Would be better to create another PR just for fixing this issue.

Reply:
Moved to #6433